gmp compiler failure on Ubuntu and webOS

On Ubuntu 25.10 I am unable to compile inputstream.ffmpegdirect:

To reproduce:
```
git clone --branch master https://github.com/xbmc/xbmc.git
git clone https://github.com/xbmc/inputstream.ffmpegdirect.git
cd inputstream.ffmpegdirect && mkdir build && cd build
cmake -DADDONS_TO_BUILD=inputstream.ffmpegdirect -DADDON_SRC_PREFIX=../.. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../../kodi-ubuntu/build/addons -DPACKAGE_ZIP=1 ../../kodi-ubuntu/cmake/addons
```

make
```
[  1%] Creating directories for 'zlib'
[  2%] Performing download step (download, verify and extract) for 'zlib'
-- verifying file...
       file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/zlib-1.3.1.tar.xz'
-- File already exists and hash match (skip download):
  file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/zlib-1.3.1.tar.xz'
  SHA256='38ef96b8dfe510d42707d9c781877914792541133e1870841463bfa73f883e32'
-- extracting...
     src='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/zlib-1.3.1.tar.xz'
     dst='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/zlib/src/zlib'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  3%] No update step for 'zlib'
[  4%] Performing patch step for 'zlib'
patching file CMakeLists.txt
patching file CMakeLists.txt
[  5%] Performing configure step for 'zlib'
CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


-- Renaming
--     /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/zlib/src/zlib/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/zlib/src/zlib-build
[  6%] Performing build step for 'zlib'
[100%] Built target zlib
[  7%] Performing install step for 'zlib'
[100%] Built target zlib
Install the project...
-- Install configuration: "Debug"
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/lib/libz.a
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/include/zconf.h
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/include/zlib.h
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/share/man/man3/zlib.3
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/lib/pkgconfig/zlib.pc
[  8%] Completed 'zlib'
[  8%] Built target zlib
[  9%] Creating directories for 'bz2'
[ 10%] Performing download step (download, verify and extract) for 'bz2'
-- verifying file...
       file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/bzip2-1.0.8.tar.gz'
-- File already exists and hash match (skip download):
  file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/bzip2-1.0.8.tar.gz'
  SHA256='ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269'
-- extracting...
     src='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/bzip2-1.0.8.tar.gz'
     dst='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/bz2/src/bz2'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 11%] No update step for 'bz2'
[ 12%] Performing patch step for 'bz2'
[ 13%] Performing configure step for 'bz2'
CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/bz2/src/bz2-build
[ 14%] Performing build step for 'bz2'
[100%] Built target bz2
[ 15%] Performing install step for 'bz2'
[100%] Built target bz2
Install the project...
-- Install configuration: "Debug"
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/lib/libbz2.a
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/include/bzlib.h
[ 15%] Completed 'bz2'
[ 15%] Built target bz2
[ 15%] Creating directories for 'gas-preprocessor'
[ 16%] Performing download step (download, verify and extract) for 'gas-preprocessor'
-- verifying file...
       file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/971df11b06a94dd42881144e4e53be30eb3fee68.tar.gz'
-- File already exists and hash match (skip download):
  file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/971df11b06a94dd42881144e4e53be30eb3fee68.tar.gz'
  SHA256='5b5a47869a45f6c33d3857b923ec409e41e0813b90b63b7e120b45e0037ee3dc'
-- extracting...
     src='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/971df11b06a94dd42881144e4e53be30eb3fee68.tar.gz'
     dst='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/gas-preprocessor/src/gas-preprocessor'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 17%] No update step for 'gas-preprocessor'
[ 18%] Performing patch step for 'gas-preprocessor'
[ 19%] Performing configure step for 'gas-preprocessor'
CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/gas-preprocessor/src/gas-preprocessor-build
[ 20%] Performing build step for 'gas-preprocessor'
[ 21%] Performing install step for 'gas-preprocessor'
Install the project...
-- Install configuration: "Debug"
-- Up-to-date: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/depends/bin/gas-preprocessor.pl
[ 22%] Completed 'gas-preprocessor'
[ 22%] Built target gas-preprocessor
[ 23%] Creating directories for 'iconv'
[ 24%] Performing download step (download, verify and extract) for 'iconv'
-- verifying file...
       file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/libiconv-1.18.tar.gz'
-- File already exists and hash match (skip download):
  file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/libiconv-1.18.tar.gz'
  SHA256='3b08f5f4f9b4eb82f151a7040bfd6fe6c6fb922efe4b1659c66ea933276965e8'
-- extracting...
     src='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/libiconv-1.18.tar.gz'
     dst='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/iconv/src/iconv'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 25%] No update step for 'iconv'
[ 26%] Performing patch step for 'iconv'
patching file lib/converters.h
Hunk #1 succeeded at 130 (offset 8 lines).
patching file lib/encodings.def
patching file lib/utf8mac.h
patching file src/Makefile.in
[ 26%] Performing configure step for 'iconv'
CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/iconv/src/iconv-build
[ 27%] Performing build step for 'iconv'
[100%] Built target iconv
[ 28%] Performing install step for 'iconv'
[100%] Built target iconv
Install the project...
-- Install configuration: "Debug"
make[4]: *** No rule to make target 'install-lib'.  Stop.
[ 29%] Completed 'iconv'
[ 29%] Built target iconv
[ 30%] Creating directories for 'gmp'
[ 31%] Performing download step (download, verify and extract) for 'gmp'
-- verifying file...
       file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/gmp-6.3.0.tar.xz'
-- File already exists and hash match (skip download):
  file='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/gmp-6.3.0.tar.xz'
  SHA256='a3c2b80201b89e68616f4ad30bc66aee4927c3ce50e33929ca819d5c43538898'
-- extracting...
     src='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/download/gmp-6.3.0.tar.xz'
     dst='/home/cscd98/Developer/inputstream.ffmpegdirect/build/build/gmp/src/gmp'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 32%] No update step for 'gmp'
[ 33%] Performing patch step for 'gmp'
[ 34%] Performing configure step for 'gmp'
CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Warning:
  Ignoring extra path from command line:

   "/home/cscd98/Developer/kodi-ubuntu/build/addons"


CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/cscd98/Developer/inputstream.ffmpegdirect/build/build/gmp/src/gmp-build
[ 35%] Performing build step for 'gmp'
[ 12%] Performing configure step for 'gmp'
checking build system type... zen3-pc-linux-gnu
checking host system type... zen3-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking ABI=64
checking compiler gcc -O2 -pedantic -fomit-frame-pointer -m64 ... no, long long reliability test 1
checking whether cc is gcc... yes
checking compiler cc -O2 -pedantic -fomit-frame-pointer -m64 ... no, long long reliability test 1
checking ABI=x32
checking compiler gcc -O2 -pedantic -fomit-frame-pointer -mx32 ... no, program does not run
checking whether cc is gcc... yes
checking compiler cc -O2 -pedantic -fomit-frame-pointer -mx32 ... no, program does not run
checking ABI=32
checking compiler gcc -m32 -O2 -pedantic -fomit-frame-pointer ... no, long long reliability test 1
checking compiler gcc -O2 -pedantic -fomit-frame-pointer ... no, long long reliability test 1
checking compiler icc -no-gcc ... no
checking whether cc is gcc... yes
checking compiler cc -m32 -O2 -pedantic -fomit-frame-pointer ... no, long long reliability test 1
checking compiler cc -O2 -pedantic -fomit-frame-pointer ... no, long long reliability test 1
configure: error: could not find a working compiler, see config.log for details
make[5]: *** [CMakeFiles/gmp.dir/build.make:92: gmp-prefix/src/gmp-stamp/gmp-configure] Error 1
make[4]: *** [CMakeFiles/Makefile2:87: CMakeFiles/gmp.dir/all] Error 2
make[3]: *** [Makefile:136: all] Error 2
make[2]: *** [CMakeFiles/gmp.dir/build.make:86: build/gmp/src/gmp-stamp/gmp-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:341: CMakeFiles/gmp.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

Seems to fail setting to ABI=64 only:

```
cd /home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/gmp/src/gmp-build/gmp-prefix/src/gmp-build && /home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/gmp/src/gmp/configure --prefix=/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/depends --disable-assembly --enable-static --disable-shared --with-pic ABI=64
checking build system type... zen3-pc-linux-gnu
checking host system type... zen3-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking ABI=64
checking compiler gcc -O2 -pedantic -fomit-frame-pointer -m64 ... no, long long reliability test 1
checking whether cc is gcc... yes
checking compiler cc -O2 -pedantic -fomit-frame-pointer -m64 ... no, long long reliability test 1
configure: error: could not find a working compiler, see config.log for details
make[5]: *** [CMakeFiles/gmp.dir/build.make:92: gmp-prefix/src/gmp-stamp/gmp-configure] Error 1
make[5]: Leaving directory '/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/gmp/src/gmp-build'
make[4]: *** [CMakeFiles/Makefile2:87: CMakeFiles/gmp.dir/all] Error 2
make[4]: Leaving directory '/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/gmp/src/gmp-build'
make[3]: *** [Makefile:136: all] Error 2
make[3]: Leaving directory '/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu/build/gmp/src/gmp-build'
make[2]: *** [CMakeFiles/gmp.dir/build.make:86: build/gmp/src/gmp-stamp/gmp-build] Error 2
make[2]: Leaving directory '/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu'
make[1]: *** [CMakeFiles/Makefile2:341: CMakeFiles/gmp.dir/all] Error 2
make[1]: Leaving directory '/home/cscd98/Developer/inputstream.ffmpegdirect/build-ubuntu'
make: *** [Makefile:91: all] Error 2
```

--- a/depends/common/gmp/0001-gmp-gcc-15-updated.patch
+++ b/depends/common/gmp/0001-gmp-gcc-15-updated.patch
@@ -1,0 +1,21 @@
+
+# HG changeset patch
+# User Marc Glisse <marc.glisse@inria.fr>
+# Date 1738186682 -3600
+# Node ID 8e7bb4ae7a18b1405ea7f9cbcda450b7d920a901
+# Parent  e84c5c785bbe8ed8c3620194e50b65adfc2f5d83
+Complete function prototype in acinclude.m4 for C23 compatibility
+
+diff -r e84c5c785bbe -r 8e7bb4ae7a18 acinclude.m4
+--- a/acinclude.m4	Wed Dec 04 18:26:27 2024 +0100
++++ b/acinclude.m4	Wed Jan 29 22:38:02 2025 +0100
+@@ -609,7 +609,7 @@
+ 
+ #if defined (__GNUC__) && ! defined (__cplusplus)
+ typedef unsigned long long t1;typedef t1*t2;
+-void g(){}
++void g(int a,t1 const*b,t1 c,t2 d,t1 const*e,int f){}
+ void h(){}
+ static __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
+ {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
+

--- a/depends/common/gmp/CMakeLists.txt
+++ b/depends/common/gmp/CMakeLists.txt
@@ -27,6 +27,7 @@ if(CORE_SYSTEM_NAME STREQUAL osx OR
 
   externalproject_add(gmp
                       SOURCE_DIR ${CMAKE_SOURCE_DIR}
+                      PATCH_COMMAND autoreconf -vif
                       CONFIGURE_COMMAND
                           bash --login -c
                           "./configure \

